src/libs/ajax/loaded-data/useLoadedData.test.ts

```diff
@@ -0,0 +1,182 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+import { useLoadedData, UseLoadedDataResult } from 'src/libs/ajax/loaded-data/useLoadedData'
+import LoadedState from 'src/libs/type-utils/LoadedState'
+import { delay } from 'src/libs/utils'
+
+
+interface TestData {
+  propA: string
+  propB: number
+}
+
+describe('useLoadedData hook', () => {
+  it('handles initial none state', () => {
+    // Act
+    const hookRender = renderHook(() => useLoadedData<TestData>())
+    const hookResult: UseLoadedDataResult<TestData> = hookRender.result.current
+
+    // Assert
+    const expectedState: LoadedState<TestData> = { status: 'None' }
+    expect(hookResult[0]).toEqual(expectedState)
+  })
+
+  it('handles loading, then ready state', async () => {
+    // Act
+    const hookRender = renderHook(() => useLoadedData<TestData>())
+    const hookResult1: UseLoadedDataResult<TestData> = hookRender.result.current
+    const updateData = hookResult1[1]
+    void updateData(async (): Promise<TestData> => {
```

Comment:
Wrap in act(() => { void updateData... })

This will get rid of the junk in the console output of running the test.
There are two other spots I'll point out that could use the same fix.

Reply:
ugg... I had fixed this but intelliJ did a strange thing that seamed to knock the file version back a few minutes.

Reply:
fixed